service/src/main/kotlin/fi/espoo/evaka/varda/integration/MockVardaIntegrationEndpoint.kt

```diff
@@ -32,177 +36,226 @@ private val logger = KotlinLogging.logger {}
 @RestController
 @RequestMapping("/mock-integration/varda/api")
 class MockVardaIntegrationEndpoint(private val mapper: ObjectMapper) {
-    val decisions = mutableListOf<VardaDecision>()
-    val feeData = mutableListOf<VardaFeeData>()
-    val children = mutableListOf<VardaChildRequest>()
+    private val mutex = Mutex()
+    private fun <T> withMockDataLock(fn: () -> T): T = runBlocking {
+        mutex.withLock {
+            fn()
+        }
+    }
+
+    var organizerId = 0L
+    var unitId = 0L
+    var personId = 0L
+    val people = mutableMapOf<Long, VardaPersonRequest>()
+    var decisionId = 0L
+    val decisions = mutableMapOf<Long, VardaDecision>()
+    var placementId = 0L
+    val placements = mutableMapOf<Long, VardaPlacement>()
+    var feeDataId = 0L
+    val feeData = mutableMapOf<Long, VardaFeeData>()
+    var childId = 0L
+    val children = mutableMapOf<Long, VardaChildRequest>()
+
+    fun cleanUp() {
```

Comment:
`cleanUp` should also use the lock, because it mutates global state

Reply:
Oh yeah, true!